xdsl/ir/core.py

```diff
@@ -536,21 +536,52 @@ class Use:
     """The index of the operand using the value in the operation."""
 
 
+@dataclass
+class IRUses(Iterable[Use]):
+    """
+    Multi-pass iterable of the uses of an IR value (SSAValue or Block).
+    """
+
+    ir: IRWithUses
+
+    def __iter__(self):
+        return self.ir._uses.__iter__()  # pyright: ignore[reportPrivateUsage]
+
+    def __len__(self):
+        return len(self.ir._uses)  # pyright: ignore[reportPrivateUsage]
+
+    def __bool__(self) -> bool:
+        """Returns `True` if there are operations in this block."""
+        return bool(self.ir._uses)  # pyright: ignore[reportPrivateUsage]
+
+    def copy(self) -> Iterable[Use]:
+        """
+        Returns a copy of the uses.
+        This is useful to avoid modifying the original uses while iterating over them.
+        """
+        return self.ir._uses.copy()  # pyright: ignore[reportPrivateUsage]
```

Comment:
Is this correct? Does the output not need to be wrapped in some sort of `IRUses` object?

Reply:
TBH it might be more straightforward and more like the end result to let clients build a tuple themselves rather than call copy

Reply:
Fine by me, but should I do this as a last commit in the stack, so this commit does not change any other code?

Reply:
I think it would be nicer to do here but if it's a tonne of work then I don't mind it being at the end

Reply:
I added this in that PR